docs/developer/best_practices/commit.rst

```diff
@@ -0,0 +1,356 @@
+and `#1831 <https://github.com/scality/metalk8s/issues/1831>`_ (because they are similar) and `#839 <https://github.com/scality/metalk8s/issues/839>`_ (because it was making the other
+changes easier), but it uses distincts commits for each issue.
+
+How to write a commit message
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Why do we need commit message
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After comments in the code, commit messages are the easiest way to find context
+for every single line of code: running ``git blame`` on a file will give you,
+for each line, the identifier of the last commit that changed the line.
+
+Unlike a comment in the code (which applies to a single line or file), a commit
+message applies to a logical change and thus can provide information on the
+design of the code and why the change was done. This makes commit messages a
+part of the code documentation and makes them helpful for other developers to
+understand your code.
+
+Last but not least: commit messages can also be used for automating tasks such
+as issue management and changelog generation.
+
+Note that it is important to have all the necessary information in the commit
+message, instead of having them (only) in the related issue, because:
+
+- the issue can contain troubleshooting/design discussion/investigation with a
+  lot of back and forth, which makes hard to get the gist of it.
+
+- you need access to an external service to get the whole context, which goes
+  against one of biggest advantage of the distributed SCM (having all the
+  information you need offline, from your local copy of the repository).
+
+Anatomy of a good commit message
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A commit is composed of a subject, a body and a footer. A blank line separates
+the subject from body and the body from the footer.
+
+The body can be omitted for trivial commit. That being said, be very careful:
+a change might seem trivial when you write it but will seem totally awkward
+the day you will have to understand why you made it. If you think your patch is
+trivial and somebody tells you he does not understand your patch, then your
+patch is not trivial and it requires a detailed description.
+
+The footer contains references for issue management (``Refs``, ``Closes``, …)
+or other relevant annotations (cherry-pick source, …).
+Optional if your commit is not related to any issue (should be pretty rare).
+
+Subject
+:::::::
+
+A good commit message should start with a short summary of the change: the
+subject line.
+
+This summary should be written using the imperative mood and carry as much
+information as possible while staying short, ideally under 50 characters (this
+is a goal, the hard limit is 72).
+
+Do not end the subject line with a period.
+
+It is composed of:
+
+- a topic, usually the name of the affected component (``ui``, ``build``,
+  ``docs``, …)
+
+- a slash and then the name of the sub-component (optional)
+
+- a colon
+
+- the description of the change
```

Comment:
- no capital

Reply:
:+1: 

And since subject line are more like a title, they shouldn't end with a period (I'll remove that part).